table.go

```diff
@@ -66,3 +71,66 @@ func (cc *Conn) AddTable(t *Table) *Table {
 	})
 	return t
 }
+
+// ListTables returns currently configured tables in the kernel
+func (cc *Conn) ListTables() ([]*Table, error) {
+	conn, err := cc.dialNetlink()
+	if err != nil {
+		return nil, err
+	}
+
+	defer conn.Close()
+
+	msg := netlink.Message{
+		Header: netlink.Header{
+			Type:  netlink.HeaderType((unix.NFNL_SUBSYS_NFTABLES << 8) | unix.NFT_MSG_GETTABLE),
+			Flags: netlink.Request | netlink.Dump,
+		},
+		Data: extraHeader(uint8(unix.AF_UNSPEC), 0),
+	}
+
+	response, err := conn.Execute(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	var tables []*Table
+	for _, m := range response {
+		t, err := tableFromMsg(m)
+		if err != nil {
+			return nil, err
+		}
+
+		tables = append(tables, t)
+	}
+
+	return tables, nil
+}
+
+func tableFromMsg(msg netlink.Message) (*Table, error) {
+	if got, want := msg.Header.Type, tableHeaderType; got != want {
+		return nil, fmt.Errorf("unexpected header type: got %v, want %v", got, want)
+	}
+
+	var t Table
+	t.Family = TableFamily(msg.Data[0])
+
+	ad, err := netlink.NewAttributeDecoder(msg.Data[4:])
+	if err != nil {
+		return nil, err
+	}
+
+	for ad.Next() {
+		switch ad.Type() {
+		case unix.NFTA_TABLE_NAME:
+			t.Name = ad.String()
+			/*
```

Comment:
What about this commented out code? Is there anything left to do here (if so, can you implement it?) or is this cruft?

Reply:
They are attributes returned from the kernel that are not yet represented on the Table struct.  I found them when decoding kernel responses- but don't yet have a need for them.  I put them (commented out) on the decoding function as a reminder but forgot they were there when submitting the PR.

I will remove them for now; unless you'd prefer to extend the structs to represent the attributes for completeness?

Reply:
If they’re easy enough to test, let’s do completeness to save the person needing them some work :) 

If not, you can leave them out.